src/commands/test.js

```diff
@@ -1,525 +1,63 @@
-      }
-    }
-
-    if (result.exitCode == 0) {
-      toolbox.print.success('✔ Tests passed')
-    } else {
-      toolbox.print.error('✖ Tests failed')
-    }
-
-    // Capture logs
-    nodeLogs =
-      nodeLogs || result.exitCode !== 0
-        ? await collectGraphNodeLogs(tempdir, standaloneNode, nodeOutputChunks)
-        : undefined
-    ethereumLogs = ethereumLogs ? await collectEthereumLogs(tempdir) : undefined
-
-    // Bring down the test environment
-    try {
-      await stopTestEnvironment(tempdir)
-    } catch (e) {
-      // do nothing (the spinner already logs the problem)
-    }
-
-    if (nodeLogs) {
-      toolbox.print.info('')
-      toolbox.print.info('  Graph node')
-      toolbox.print.info('  ----------')
-      toolbox.print.info('')
-      toolbox.print.info(indent('  ', nodeLogs))
+    const platform = getPlatform();
+    if (!version) {
+      let xmlHttp = new XMLHttpRequest();
+      xmlHttp.open('GET', 'https://api.github.com/repos/LimeChain/matchstick/releases/latest', false);
+      xmlHttp.send();
+      const jsonResponse = JSON.parse(xmlHttp.responseText);
+      version = jsonResponse.tag_name;
     }
 
-    if (ethereumLogs) {
-      toolbox.print.info('')
-      toolbox.print.info('  Ethereum')
-      toolbox.print.info('  --------')
-      toolbox.print.info('')
-      toolbox.print.info(indent('  ', ethereumLogs))
-    }
-
-    // Always print the test output
-    toolbox.print.info('')
-    toolbox.print.info('  Output')
-    toolbox.print.info('  ------')
-    toolbox.print.info('')
-    toolbox.print.info(indent('  ', result.output))
+    const url = `https://github.com/LimeChain/matchstick/releases/download/${ version }/${ platform }`;
 
-    // Propagate the exit code from the test run
-    process.exitCode = result.exitCode
-  },
+    let binary = new Binary(platform, url, version);
+    await binary.install(force);
+    binary.run(datasource);
+  }
 }
 
-/**
- * Indents all lines of a string
- */
-const indent = (indentation, str) =>
-  str
-    .split('\n')
-    .map(s => `${indentation}${s}`)
-    // Remove whitespace from empty lines
-    .map(s => s.replace(/^\s+$/g, ''))
-    .join('\n')
-
-const configureTestEnvironment = async (toolbox, tempdir, composeFile, nodeImage) =>
-  await withSpinner(
-    `Configure test environment`,
-    `Failed to configure test environment`,
-    `Warnings configuring test environment`,
-    async spinner => {
-      // Temporary compose file
-      let tempComposeFile = path.join(tempdir, 'compose', 'docker-compose.yml')
+function getPlatform() {
+  const type = os.type();
+  const arch = os.arch();
 
-      // Copy the compose file to the temporary directory
-      await toolbox.filesystem.copy(composeFile, tempComposeFile)
+  if (type === 'Windows_NT' && arch === 'x64') return 'binary-windows';
+  if (type === 'Linux' && arch === 'x64') return 'binary-linux';
+  if (type === 'Darwin' && arch === 'x64') return 'binary-macos';
```

Comment:
Maybe you could aggregate all if's related to `arch` in a single one like this, so lines are shorter:

```js
if (arch === 'x64') {
  if (type === 'Windows_NT') return 'binary-windows';
  if (type === 'Linux') return 'binary-linux';
  if (type === 'Darwin') return 'binary-macos'
}
```

This is more of prefference, you don't have to change if you don't want 🙂